src/components/oculus-touch-controls.js

```diff
@@ -103,50 +98,49 @@ module.exports.Component = registerComponent('oculus-touch-controls', {
     el.removeEventListener('buttonup', this.onButtonUp);
     el.removeEventListener('touchstart', this.onButtonTouchStart);
     el.removeEventListener('touchend', this.onButtonTouchEnd);
+    el.removeEventListener('axismove', this.onAxisMoved);
     el.removeEventListener('model-loaded', this.onModelLoaded);
   },
 
   checkIfControllerPresent: function () {
     var data = this.data;
-    var isPresent = this.isControllerPresent(this.el.sceneEl, GAMEPAD_ID_PREFIX, { hand: data.hand });
+    var isPresent = false;
+    // Find which controller matches both prefix and hand.
+    var whichControllers = (this.getGamepadsByPrefix(GAMEPAD_ID_PREFIX) || [])
+      .filter(function (gamepad) { return gamepad.hand === data.hand; });
+    if (whichControllers && whichControllers.length) { isPresent = true; }
     if (isPresent === this.controllerPresent) { return; }
     this.controllerPresent = isPresent;
     if (isPresent) {
-      this.injectTrackedControls(); // inject track-controls
+      // Inject with specific gamepad id, if provided. This works around a temporary issue
+      // where Chromium uses `Oculus Touch (Right)` but Firefox uses `Oculus Touch (right)`.
+      this.injectTrackedControls(whichControllers[0]);
       this.addEventListeners();
-    } else {
-      this.removeEventListeners();
-    }
-  },
-
-  onGamepadConnected: function (evt) {
-    // for now, don't disable controller update listening, due to
-    // apparent issue with FF Nightly only sending one event and seeing one controller;
-    // this.everGotGamepadEvent = true;
-    // this.removeControllersUpdateListener();
-    this.checkIfControllerPresent();
+    } else { this.removeEventListeners(); }
   },
 
-  onGamepadDisconnected: function (evt) {
-    // for now, don't disable controller update listening, due to
-    // apparent issue with FF Nightly only sending one event and seeing one controller;
-    // this.everGotGamepadEvent = true;
-    // this.removeControllersUpdateListener();
+  onGamepadConnectionEvent: function (evt) {
```

Comment:
We can get rid of this method and do `window.addEventListener('gamepaddisconnected', this.checkIfControllerPresent, false);`